marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flows/provenance/sourceProvenance.sjs

```diff
@@ -0,0 +1,78 @@
+    "The database 'data-hub-STAGING' should be captured."
+  ));
+  assertions.push(test.assertEqual("job:my-job-id", fn.string(provDocument.xpath("/*:document/*:wasGeneratedBy/*:activity/@*:ref")),
+    "The document should be generated by 'job:my-job-id'."
+  ));
+  assertions.push(test.assertEqual("step:myStep", fn.string(provDocument.xpath("/*:document/*:wasInfluencedBy/*:influencer/@*:ref")),
+    "The document should be influenced by 'step:myStep'."
+  ));
+  assertions.push(test.assertEqual(`user:${xdmp.getCurrentUser()}`, fn.string(provDocument.xpath("/*:document/*:wasAttributedTo/*:agent/@*:ref")),
+    `The document should have wasAttributedTo 'user:${xdmp.getCurrentUser()}'.`
+  ));
+  assertions.push(test.assertEqual('external:customers.csv', fn.substringBefore(fn.string(provDocument.xpath("/*:document/*:wasDerivedFrom/*:usedEntity/@*:ref")), "#"),
+    `The document should have wasDerivedFrom 'external:customers.csv'.`
+  ));
+}
+
+function assertValidSourceProvenance(provDocument) {
+  assertions.push(test.assertEqual("customers.csv", fn.string(provDocument.xpath("/*:document/*:entity/file")),
+    "The file 'customers.csv' should be captured."
+  ));
+  assertions.push(test.assertEqual("Oracle", fn.string(provDocument.xpath("/*:document/*:entity/dataSourceName")),
+    "The data source name 'Oracle' should be captured."
+  ));
+  assertions.push(test.assertEqual("SQL Database", fn.string(provDocument.xpath("/*:document/*:entity/dataSourceType")),
+    "The data source type 'SQL Database' should be captured."
+  ));
+}
+
+const flowName = "doesntMatter";
+
+const fakeFlow = {
+  name:flowName,
+  steps: {
+    "1": {
+      name:"myStep",
+      stepDefinitionName: "default-ingestion",
+      stepDefinitionType: "ingestion"
+    }
+  }
+};
+
+let provDocuments;
+let stepExecutionContext = new StepExecutionContext(fakeFlow, "1", {name:"default-ingestion", type: "ingestion"}, "my-job-id", { latestProvenance: true, file: "customers.csv", sourceName: "Oracle", sourceType:"SQL Database", targetDatabase: "data-hub-STAGING" });
+
+// test JSON Object entity instance
+stepExecutionContext.completedItems = ["testJSONObject.json"];
+flowProvenance.writeProvenanceData(stepExecutionContext, [
+  {uri: "testJSONObject.json", value: { envelope:{ instance: {}}}}
+]);
+
+
+provDocuments = hubTest.getProvDocuments();
+assertions.push(test.assertEqual(2, fn.count(provDocuments),
+  "Two provenance documents should be captured (1 source + 1 written record)."
+));
+for (let provDocument of provDocuments) {
+  if (fn.exists(provDocument.xpath("/*:document/*:entity/*:dataSourceName"))) {
+    assertValidSourceProvenance(provDocument);
+  } else {
+    assertValidRecordProvenance(provDocument);
+    assertions.push(test.assertEqual("testJSONObject.json", fn.string(provDocument.xpath("*:document/*:entity/documentURI")),
+      "The document URI 'testJSONObject.json' should be captured."
+    ));
+  }
+}
+hubTest.clearJobsDatabase();
```

Comment:
I don't think you need to do this here - it's usually nice for a test to leave its data behind in case it fails, as that makes debugging easier. It's the responsibility of each test to clear any databases it cares about before it runs.